ui/app/src/utils/content.ts

```diff
@@ -228,41 +228,31 @@ function getLegacyItemSystemType(item: LegacyItem): SystemType {
 	}
 }
 
-export function parseLegacyItemToBaseItem(item: LegacyItem): BaseItem {
+export function parseLegacyItemToContentItem(item: LegacyItem): ContentItem;
+export function parseLegacyItemToContentItem(item: LegacyItem[]): ContentItem[];
+export function parseLegacyItemToContentItem(item: LegacyItem | LegacyItem[]): ContentItem | ContentItem[] {
+	if (Array.isArray(item)) {
+		// If no internalName then skipping (e.g. level descriptors)
+		return item.flatMap((i) => (i.internalName || i.name ? [parseLegacyItemToContentItem(i)] : []));
+	}
+
 	const stateMap = getStateMapFromLegacyItem(item);
 	const state = getStateBitmap(stateMap);
 	return {
-		id: item.uri ?? item.path,
+		path: item.uri ?? item.path,
 		label: item.internalName ?? item.name,
+		systemType: getLegacyItemSystemType(item),
+		mimeType: item.mimeType,
+		id: item.uri ?? item.path,
 		parentId: null,
 		contentTypeId: item.contentType,
-		path: item.uri ?? item.path,
 		// Assuming folders aren't navigable
 		previewUrl: item.uri?.includes('index.xml') ? item.browserUri || '/' : null,
-		systemType: getLegacyItemSystemType(item),
-		mimeType: item.mimeType,
 		state,
 		stateMap,
 		lockOwner: null,
-		disabled: null,
 		localeCode: 'en',
 		translationSourceId: null,
-		availableActions: null,
-		availableActionsMap: null,
-		childrenCount: 0
```

Comment:
- Please restore order of properties to avoid unnecessary diff
- Why are availableActions, availableActionsMap, childrenCount and disabled removed?

Reply:
The detail here is that the diff is a bit messy because there are 3 utils merged into one in here:
- parseLegacyItemToBaseItem
- parseLegacyItemToSandboxItem
- parseLegacyItemToDetailedItem
Those 3 are now `parseLegacyItemToContentItem`.
I tried matching the fields order between the 3 utils, and availableActions, availableActionsMap, childrenCount are there. Disabled is not a prop of the ContentItem anymore (is a prop of the state/stateMap fields only now).